pkg/conf/conf.go

```diff
@@ -156,6 +156,15 @@ type Main struct {
 	// Histogram parameters for the processing duration
 	ProcessingDurationBucketFactor float64
 	ProcessingDurationBuckets      int
+
+	// RateLimiter config
+	RateLimiterEnabled                  bool
+	RateLimiterContainerRecordLimit     int
+	RateLimiterGlobalRecordLimit        int
+	RateLimiterWindowSizeSec            int
+	RateLimiterIntervalMs               int
+	RateLimiterPerReaderRecordThreshold int
```

Comment:
I have several reservations about using record threshold to check the limiter:

1. Are there situations where we actually need to check the limiter by record threshold and not by the timer tick? 

2. This would be much more useful if we could do it globally as pointed out in the PR description: "the number of records not added to rate limiter has become more than a threshold". It's clear, however, that we cannot have a global threshold without introducing a global lock.

3. I'm not sure it is a good idea to have two parameters to tune the limiter. This would make tuning significantly harder and future investigations significantly more complex.

I think we need to consider skipping this parameter and related mechanism for now. If we see that it is necessary, we can add it.

Reply:
During our discussion, @emadolsky mentioned that this mechanism is necessary to prevent the limiter from stalling in certain situations. With that in mind, we will have to keep it.